src/components/workspace/BuildProject/BuildProject.tsx

```diff
@@ -638,9 +644,29 @@ const BuildProject: FC<Props> = ({ projectId, contract, updateContract }) => {
     if (activeProject?.network) {
       setEnvironment(activeProject.network);
     }
-    if (activeProject?.selectedContract) {
-      setSelectedContract(activeProject.selectedContract);
-      deployForm.setFieldsValue({ contract: activeProject.selectedContract });
+    const selectedABIPath = activeProject?.selectedContract;
+
+    if (selectedABIPath) {
+      const correspondingScriptPath = selectedABIPath.replace('.abi', '.ts');
+
+      let _selectedContract: string | undefined = selectedABIPath;
```

Comment:
This variable doesn't have to be mutable. By abstracting the logic into `getSelectedContract()` it could be just a `return`.